Start documenting prioritary tasks. (#32)
ApolloTeam-dev/AROS: docs/tasks/GFX-subsystem.md

# GFX subsystem

> Source code:

Reviewer: ```suggestion
## Source code
```

> - [`rom/graphics/`](https://github.com/ApolloTeam-dev/AROS/tree/master-new/rom/graphics/)

Reviewer: ```suggestion
- [`rom/graphics/`](https://github.com/ApolloTeam-dev/AROS/tree/master/rom/graphics/)
```

> Documentation from AROS:

Reviewer: ```suggestion
## Documentation from AROS
```

## Issues

Amiga OS RTG subsystems usually have conversion routines allowing them to blit from any format to any format: blit from 8bit to 16bit, blit from 16bit to 24bit, etc.

Aros lacks some of those and falls back to “for each pixel, read it single and cast it single to new format” which is about 1000 times slower than the normal bulk blit routines you would expect on Amiga RTG.

It’s visually easy to see if you hit this on Aros because they render from bottom to top and not from top to bottom as usual.

Fixing these render limits would make a major improvement to Aros... as Aros with an 68020 would sometimes take basically *minutes* (68080 is roughly 100 times faster than A1200) to refresh a frame.

You can trigger this easily:
1. Boot without startup sequence: go in early boot, select boot with no startup.
2. In CLI type `workbook`.
3. Click on some HD icon so that the window appears.
4. Open a few folder windows and move them around a little, see how fast it is.

High priority bugs for the GFX subsystem are listed under:
https://github.com/ApolloTeam-dev/AROS/issues/8

## Compiling and testing
Workbook / `workbench.library` is in ROM.

> - With the [`master-new`](https://github.com/ApolloTeam-dev/AROS) branch, which is the default, use `build_all.sh`.

Reviewer: ```suggestion
- With the [`master`](https://github.com/ApolloTeam-dev/AROS) branch, which is the default, use `build_all.sh`.
```

> - If you are using the [`mkapollo.sh`](https://github.com/ApolloTeam-dev/AROS/blob/v4-alynna/mkapollo.sh) script, `./mkapollo.sh kernel` suffices to build it. There are some improvements pending merge at: [Some mkapollo.sh enhancements #27](https://github.com/ApolloTeam-dev/AROS/pull/27)

Reviewer: mkapollo is merged to master now
```suggestion
- If you are using the [`mkapollo.sh`](https://github.com/ApolloTeam-dev/AROS/blob/master/mkapollo.sh) script, `./mkapollo.sh kernel` suffices to build it. 
```
